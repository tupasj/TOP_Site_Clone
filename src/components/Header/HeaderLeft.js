const HeaderLeft = () => {
  return (
    <>
    <span>HeaderLeft</span>
    </>
  )
};
export { HeaderLeft };
