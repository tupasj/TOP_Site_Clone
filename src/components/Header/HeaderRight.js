const HeaderRight = () => {
    return (
        <>
        <span>HeaderRight</span>
        </>
    )
}

export { HeaderRight };
