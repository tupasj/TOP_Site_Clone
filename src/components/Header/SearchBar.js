import { useState } from "react";
import { useNavigate } from "react-router-dom";

const SearchBar = () => {
  const [searchText, setSearchText] = useState("");
  const navigate = useNavigate();

  const handleChange = (e) => {
    setSearchText(e.target.value);
  };

  const handleSubmit = (e) => {
    if (e.key === "Enter" || e.type !== "keydown") {
      navigate(`/channel/${searchText}`);
    };
  };

  return (
    <div className="searchbar">
      <div className="searchbar__container">
        <input
          className="searchbar__input"
          type="text"
          placeholder="Search streamer"
          value={searchText}
          onChange={handleChange}
          onKeyDown={handleSubmit}
        ></input>
        <div className="searchbar__icon-container" onClick={handleSubmit}>
          <i className="searchbar__icon--magnifying-glass fa-solid fa-magnifying-glass"></i>
        </div>
      </div>
    </div>
  );
};

export { SearchBar };
