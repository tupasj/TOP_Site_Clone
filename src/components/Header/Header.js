import { HeaderLeft } from "./HeaderLeft";
import { SearchBar } from "./SearchBar";
import { HeaderRight } from "./HeaderRight";

const Header = () => {
    return (
        <header>
            <HeaderLeft />
            <SearchBar />
            <HeaderRight />
        </header>
    );
};

export { Header };