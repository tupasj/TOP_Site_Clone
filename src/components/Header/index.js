import { Header } from "./Header";
import { HeaderLeft } from "./HeaderLeft";
import { HeaderRight } from "./HeaderRight";
import { SearchBar } from "./SearchBar";

export { Header, HeaderLeft, HeaderRight, SearchBar };