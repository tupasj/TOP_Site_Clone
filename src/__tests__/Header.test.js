import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { Header } from "../components/Header/Header";

test("renders correctly", () => {
  const { container } = render(
    <MemoryRouter>
      <Header />
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});