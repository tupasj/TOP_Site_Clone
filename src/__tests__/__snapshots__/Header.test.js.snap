// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <header>
    <div
      class="header-left"
    >
      <a
        class="header-left__link"
        href="/"
      >
        <i
          class="header-left__icon--robot fa-solid fa-robot"
        />
      </a>
      <a
        class="header-left__link"
        href="/"
      >
        <span
          class="header-left__text"
        >
          Following
        </span>
      </a>
      <a
        class="header-left__link"
        href="/"
      >
        <span
          class="header-left__text"
        >
          Browse
        </span>
      </a>
      <i
        class="header-left__icon--ellipsis fa-solid fa-ellipsis-vertical"
      />
    </div>
    <div
      class="searchbar"
    >
      <div
        class="searchbar__container"
      >
        <input
          class="searchbar__input"
          placeholder="Search streamer"
          type="text"
          value=""
        />
        <div
          class="searchbar__icon-container"
        >
          <i
            class="searchbar__icon--magnifying-glass fa-solid fa-magnifying-glass"
          />
        </div>
      </div>
    </div>
    <div
      class="header-right"
    >
      <button
        class="header-right__login-btn"
      >
        Log In
      </button>
      <button
        class="header-right__signup-btn"
      >
        Sign Up
      </button>
      <i
        class="header-right__icon fa-regular fa-user"
      />
    </div>
  </header>
</div>
`;
